src/allencell_ml_segmenter/thresholding/thresholding_service.py

```diff
@@ -48,83 +44,97 @@ def __init__(
+                thresh_function: Callable = AutoThreshold(
+                    self._thresholding_model.get_autothresholding_method()
+                )
+            elif self._thresholding_model.is_threshold_enabled():
+                thresh_function = self._threshold_image
+
+            # selected layers in the ui
+            selected_idx: list[int] = (
+                self._thresholding_model.get_selected_idx()
             )
-        else:
-            thresh_function = self._threshold_image
 
-        selected_idx: Optional[list[int]] = (
-            self._file_input_model.get_selected_idx()
-        )
+            for idx, layer in enumerate(layers_containing_prob_map):
+                # for selected layers, perform thresholding and display the result in the viewer
+                if idx in selected_idx:
+                    # Creating helper functions for mypy strict typing
+                    # Thresholding function
+                    def thresholding_task(
+                        layer_instance: Layer = layer,
+                    ) -> np.ndarray:
+                        # INVARIANT: a segmentation layer must have prob_map in its metadata if it came from our plugin
+                        # so we are only supporting thresholding images that are from the plugin itself.
+                        if (
+                            not isinstance(layer_instance.metadata, dict)
+                            or "prob_map" not in layer_instance.metadata
+                        ):
+                            raise ValueError(
+                                "Layer metadata must be a dictionary containing the 'prob_map' key in order to threshold."
+                            )
+                        # This thresholding task returns a binary map
+                        return thresh_function(
+                            layer_instance.metadata["prob_map"]
+                        )
 
-        if selected_idx is not None:
-            for idx in selected_idx:
-                layer: Layer = layers_containing_prob_map[idx]
-
-                # Creating helper functions for mypy strict typing
-                def thresholding_task(
-                    layer_instance: Layer = layer,
-                ) -> np.ndarray:
-                    # INVARIANT: a segmentation layer must have prob_map in its metadata if it came from our plugin
-                    # so we are only supporting thresholding images that are from the plugin itself.
-                    if (
-                        not isinstance(layer_instance.metadata, dict)
-                        or "prob_map" not in layer_instance.metadata
-                    ):
-                        raise ValueError(
-                            "Layer metadata must be a dictionary containing the 'prob_map' key in order to threshold."
+                    # On return, display the binary map that was produced from thresholding
+                    def on_return(
```

Comment:
I think this would benfit from 1 more iteration on design: remove the 'none' option.  Id like to see how that simplifies this logic